test/core/test_task_launcher.py

```diff
@@ -74,20 +87,31 @@ def test_create_launch_expire_assignments(self):
         for assignment in launcher.assignments:
             self.assertEqual(assignment.get_status(), AssignmentState.CREATED)
 
-        launcher.launch_units("dummy-url:3000")
-
-        for unit in launcher.units:
-            self.assertEqual(unit.get_db_status(), AssignmentState.LAUNCHED)
-        for assignment in launcher.assignments:
-            self.assertEqual(assignment.get_status(), AssignmentState.LAUNCHED)
-
         launcher.expire_units()
 
         for unit in launcher.units:
             self.assertEqual(unit.get_db_status(), AssignmentState.EXPIRED)
         for assignment in launcher.assignments:
             self.assertEqual(assignment.get_status(), AssignmentState.EXPIRED)
 
+    def test_launch_assignments(self):
+        """Initialize a launcher on a task run, then create the assignments"""
+        mock_data_array = self.get_mock_assignment_data_array()
+        launcher = TaskLauncher(
+            self.db, self.task_run, mock_data_array, max_num_concurrent_units=1
+        )
+        launcher.launched_units = LimitedDict(launcher.max_num_concurrent_units)
+        launcher.create_assignments()
+        launcher.launch_units("dummy-url:3000")
+
+        for unit in launcher.units:
+            if unit.get_status() == AssignmentState.LAUNCHED:
+                unit.set_db_status(AssignmentState.COMPLETED)
+                time.sleep(10)
```

Comment:
This internal sleep can be much shorter (`0.1`) if you're looping on the condition of advancing any launched units to completing until all are completed.